src/System.Windows.Forms/src/System/Windows/Forms/TabPage.cs

```diff
@@ -461,7 +485,38 @@ protected override void OnEnter(EventArgs e)
                 }
 
                 _enterFired = false;
+                KeyboardToolTipStateMachine.Instance.NotifyAboutGotFocus(this);
+            }
+        }
+
+        internal override Rectangle GetToolNativeScreenRectangle()
```

Comment:
Should it be `ToolTip`? If not - what is `Tool`?
```suggestion
        internal override Rectangle GetToolTipNativeScreenRectangle()
```

Reply:
This method overrides Control.GetToolNativeScreenRectangle() method which used in IKeyboardToolTip interface to show keyboard tooltips. Its task is to return a current control rectangle.  This method in the TabPage class returns a rectangle of a selected tab. So a tooltip will be shown near a selected tab.
> If not - what is `Tool`?

`Tool` is a selected tab.